lib/Dialect/Triton/Transforms/RewriteTensorPointer.cpp

```diff
@@ -370,9 +370,13 @@ class RewriteTensorPointerPass
     }
 
     // update rewritedInfo
+    auto opResults = op.getResults();
     unsigned oldResIdx = 0, newResIdx = 0;
     while (oldResIdx < results.size()) {
       if (!triton::isTensorPointerType(results[oldResIdx].getType())) {
+        if (opResults.size() > 0) {
+          opResults[oldResIdx].replaceAllUsesWith(newOp.getResult(newResIdx));
+        }
```

Comment:
I don't think the if is necessary since `opResults.size()` should be the same as `results.size()` so if it was empty we would never reach this point.